frontend/src/hooks/useToolbarButtons.tsx

```diff
@@ -0,0 +1,132 @@
+import { MutableRefObject, useEffect, useRef, useState } from 'react';
+import { throttle } from 'lodash';
+import ResizeObserverPolyfill from 'resize-observer-polyfill';
+import { RIGHT_PANEL_BUTTON_COUNT } from '../utils/constants';
+
+export type UseToolbarButtonsProps = {
+  timeRoomNameRef: MutableRefObject<HTMLDivElement | null>;
+  toolbarRef: MutableRefObject<HTMLDivElement | null>;
+  mediaControlsRef: MutableRefObject<HTMLDivElement | null>;
+  overflowAndExitRef: MutableRefObject<HTMLDivElement | null>;
+  rightPanelControlsRef: MutableRefObject<HTMLDivElement | null>;
+  numberOfToolbarButtons: number;
+};
+
+export type UseToolbarButtons = {
+  displayTimeRoomName: boolean;
+  centerButtonLimit: number;
+  rightButtonLimit: number;
+};
+
+/**
+ * React hook to determine which buttons should be displayed on the toolbar.
+ * @param {UseToolbarButtonsProps} props - The props for the hook
+ *  @property {MutableRefObject<HTMLDivElement | null>} toolbarRef - The ref for the Toolbar
+ *  @property {MutableRefObject<HTMLDivElement | null>} mediaControlsRef - The ref for the audio and video controls
+ *  @property {MutableRefObject<HTMLDivElement | null>} overflowAndExitRef - The ref for the overflow and exit buttons
+ *  @property {MutableRefObject<HTMLDivElement | null>} rightPanelControlsRef - The ref for the right panel buttons
+ * @returns {UseToolbarButtons} The center and right toolbar buttons' limits, and whether to display the TimeRoomNameMeetingRoom component
+ */
+const useToolbarButtons = ({
+  timeRoomNameRef,
+  toolbarRef,
+  mediaControlsRef,
+  overflowAndExitRef,
+  rightPanelControlsRef,
+  numberOfToolbarButtons,
+}: UseToolbarButtonsProps): UseToolbarButtons => {
+  const observer = useRef<ResizeObserver | undefined>();
+  const [displayTimeRoomName, setDisplayTimeRoomName] = useState<boolean>(false);
+
+  const [centerButtonLimit, setCenterButtonLimit] = useState<number>(0);
+  const [rightButtonLimit, setRightButtonLimit] = useState<number>(0);
+  const buttonWidth = 60;
+
+  useEffect(() => {
+    if (toolbarRef.current && !observer.current) {
+      const throttledSetToolbarButtons = throttle(
+        () => {
+          if (
+            !(
```

Comment:
What does this incantation do?

Reply:
It calculates the number of buttons and whether the `TimeRoomName` are to be displayed in the toolbar. We throttle it a bit since it can be a bit taxing.